docs/setup/docker.asciidoc

```diff
@@ -26,25 +26,58 @@ endif::[]
 
 ifeval::["{release-state}"!="unreleased"]
 
-To start an {es} container for development or testing, run:
-
+. Start an {es} container for development or testing:
++
 [source,sh,subs="attributes"]
 ----
 docker network create elastic
 docker pull {es-docker-image}
-docker run --name es01-test --net elastic -p 9200:9200 -p 9300:9300 -e "discovery.type=single-node" {es-docker-image}
+docker run --name es-node01 -p 9200:9200 -p 9300:9300 -it {docker-image}
 ----
-
-To start {kib} and connect it to your {es} container, run the following commands
-in a new terminal session:
-
++
+When you start {es} for the first time, the following security configuration 
+occurs automatically:
++
+* {ref}/configuring-stack-security.html#stack-security-certificates[Certificates and keys] 
+are generated for the transport and HTTP layers.
+* The Transport Layer Security (TLS) configuration settings are written to
+`elasticsearch.yml`.
+* A password is generated for the `elastic` user.
+* An enrollment token is generated for {kib}.
++
+NOTE: You might need to scroll back a bit in the terminal to view the password 
+and enrollment token.
+
+. Copy the generated password and enrollment token and save them in a secure 
+location. These values are shown only when you start {es} for the first time.
+You'll use these to enroll {kib} with your {es} cluster and log in.
++
+[NOTE]
+====
+If you need to reset the password for the `elastic` user or other
+built-in users, run the {ref}/reset-password.html[`elasticsearch-reset-password`]
+tool. To generate new enrollment tokens for {kib} or {es} nodes, run the
+{ref}/create-enrollment-token.html[`elasticsearch-create-enrollment-token`] tool.
+These tools are available in the {es} `bin` directory.
+====
+
+. In a new terminal session, start {kib} and connect it to your {es} container:
++
 [source,sh,subs="attributes"]
 ----
 docker pull {docker-image}
-docker run --name kib01-test --net elastic -p 5601:5601 -e "ELASTICSEARCH_HOSTS=http://es01-test:9200" {docker-image}
+docker run --name kib-01 --net elastic -p 5601:5601 docker.elastic.co/kibana/kibana:{version}
 ----
++
+When you start {kib}, a unique link is output to your terminal.
+
+. To access {kib}, click the generated link in your terminal.
+
+  .. In your browser, paste the enrollment token that you copied when starting
+  {es} and click the button to connect your {kib} instance with {es}.
 
-To access {kib}, go to http://localhost:5601[http://localhost:5601].
+  .. Log in to {kib} as the `elastic` user with the password that was generated
+  when you started {es}.
 
 [float]
 === Stop Docker containers
```

Comment:
I'm not sure that the "Stop Docker containers" section makes sense given the rest of the page. Perhaps at one point these instructions showed how to run the Docker containers in  [detached mode](https://phoenixnap.com/kb/docker-run-command-with-examples)? At any rate, the current instructions show how to run the Docker containers in _attached_ mode, so when you close the terminal, the container is already stopped.

So maybe we should remove that bit and rename this section to "Remove Docker containers".

Reply:
++ I like that suggestion @jportner.